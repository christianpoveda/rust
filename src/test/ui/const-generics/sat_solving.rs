// build-pass

#![feature(const_generics)] //~ WARN: incomplete

fn foo<const N: usize>(x: [(); N + N]) -> [(); 2 * N] {
    x
}

fn main() {
    let y = foo([(), (), (), ()]);
}
