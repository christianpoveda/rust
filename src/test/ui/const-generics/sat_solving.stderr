warning: the feature `const_generics` is incomplete and may cause the compiler to crash
  --> $DIR/sat_solving.rs:1:12
   |
LL | #![feature(const_generics)]
   |            ^^^^^^^^^^^^^^
   |
   = note: `#[warn(incomplete_features)]` on by default

error[E0308]: mismatched types
  --> $DIR/sat_solving.rs:4:5
   |
LL | fn foo<const N: usize>(x: [(); N + N]) -> [(); 2 * N] {
   |                                           ----------- expected `[(); _]` because of return type
LL |     x
   |     ^ expected `2 * N`, found `N + N`
   |
   = note: expected type `[(); _]`
              found type `[(); _]`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0308`.
